lib/expand/join.go

```diff
@@ -1138,11 +1138,17 @@ func (p *Peer) validateWizardState(operator ops.Operator) (*ops.Site, *ops.SiteO
 
 	switch operation.State {
 	case ops.OperationStateInstallInitiated,
-		ops.OperationStateInstallProvisioning,
-		ops.OperationStateInstallPrechecks,
+		ops.OperationStateInstallProvisioning:
+		// Consider these states for resuming the installation
+	case ops.OperationStateInstallPrechecks,
 		ops.OperationStateFailed:
 		// Consider these states for resuming the installation
-		// (including failed that puts the operation into manual mode)
+		// (including failed that puts the operation into manual mode).
+		// Be careful about including unrelated peers though
+		if thisPeerPartOfInstallState(p.AdvertiseAddr, operation.Servers) {
```

Comment:
Why is this check only performed if the operation is in the `prechecks` and `failed` states? Trying to understand whether this covers all scenarios. E.g. what about the two states above (`initiated` and `provisioning`) and `deploying`?